addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZwaveAssociationGroupInfoCommandClass.java

```diff
@@ -0,0 +1,397 @@
+
+    // list of command classes that are eligible for auto subscription.
+    @XStreamOmitField
+    private Set<CommandClass> autoCCs = ImmutableSet.of(CommandClass.DEVICE_RESET_LOCALLY, CommandClass.BATTERY,
+            CommandClass.CONFIGURATION);
+
+    /**
+     * Creates a new instance of the ZwaveAssociationGroupInfoCommandClass class.
+     *
+     * @param node
+     *            the node this command class belongs to
+     * @param controller
+     *            the controller to use
+     * @param endpoint
+     *            the endpoint this Command class belongs to
+     */
+    public ZwaveAssociationGroupInfoCommandClass(ZWaveNode node, ZWaveController controller, ZWaveEndpoint endpoint) {
+        super(node, controller, endpoint);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public CommandClass getCommandClass() {
+        return CommandClass.ASSOCIATION_GROUP_INFO;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws ZWaveSerialMessageException
+     */
+    @Override
+    public void handleApplicationCommandRequest(SerialMessage serialMessage, int offset, int endpointId)
+            throws ZWaveSerialMessageException {
+        logger.debug("NODE {}: Received ASSOCIATION_GROUP_INFO command V{}", getNode().getNodeId(), getVersion());
+        int command = serialMessage.getMessagePayloadByte(offset);
+        switch (command) {
+            case GROUP_NAME_REPORT:
+                logger.trace("NODE {}: Process Group Name Report", getNode().getNodeId());
+                processGroupNameReport(serialMessage, offset);
+                break;
+            case INFO_REPORT:
+                logger.trace("NODE {}: Process Group Info Report", getNode().getNodeId());
+                processInfoReport(serialMessage, offset);
+                break;
+            case COMMAND_LIST_REPORT:
+                logger.trace("NODE {}: Process Group Command List Report", getNode().getNodeId());
+                processCommandListReport(serialMessage, offset);
+                break;
+            default:
+                logger.warn("NODE {}: Unsupported Command {} for command class {} ({}).", getNode().getNodeId(),
+                        command, getCommandClass().getLabel(), getCommandClass().getKey());
+
+        }
+    }
+
+    private void processGroupNameReport(SerialMessage serialMessage, int offset) throws ZWaveSerialMessageException {
+        int groupIdx = serialMessage.getMessagePayloadByte(offset + 1);
+        int numBytes = serialMessage.getMessagePayloadByte(offset + 2);
+
+        if (numBytes < 0) {
+            logger.error("NODE {}: Group Name report error in message length ({})", getNode().getNodeId(),
+                    serialMessage.getMessagePayload().length);
+        }
+
+        if (numBytes == 0) {
+            getGroupInfo(groupIdx).setName("");
+            return;
+        }
+
+        // Maximum length is 16 bytes
```

Comment:
I suspect that’s about the max as from memory the ZWave frames are about 32 bytes long…

I’d either update the comment to be inline with the definition in this file (i.e. 25) or remove the number and just say // Check maximum length